codalab/apps/web/tasks.py

```diff
@@ -781,5 +793,191 @@ def do_phase_migrations():
+                for data_type in data_types:
+                    if hasattr(phase, data_type):
+                        data_field = getattr(phase, data_type)
+                        if data_field.file.name not in file_cache.keys():
+                            logger.info("Datafield is {}".format(data_field))
+                            file_name = "{}_{}.zip".format(data_type, phase.phasenumber)
+                            phase_dict[data_type] = file_name
+                            file_cache[data_field.file.name] = {
+                                # 'phasenumber': phase.phasenumber
+                                'name': file_name
+                            }
+                            zip_file.writestr(file_name, data_field.read())
+                        else:
+                            file_name = file_cache[data_field.file.name]['name']
+                            phase_dict[data_type] = file_name
+            except ValueError:
+                logger.info("Failed to retrieve the file.")
+            # Datasets
+            datasets = phase.datasets.all()
+            if datasets:
+                phase_dict['datasets'] = dict()
+                temp_comp_dump.status = "Adding datasets"
+                temp_comp_dump.save()
+                logger.info("Adding dataset")
+                for data_set_index, data_set in enumerate(datasets):
+                    phase_dict['datasets'][data_set_index] = {
+                        'name': data_set.datafile.name,
+                        'url': data_set.datafile.source_url,
+                        'description': data_set.description
+                    }
+            # Save the phase_dict
+            yaml_data['phases'][index] = phase_dict
+        # Leaderboard
+        # main Leaderboard dict
+        yaml_data["leaderboard"] = dict()
+        # leaderboards dict
+        logger.info("Adding leaderboard.")
+        leaderboards_dict = dict()
+        for index, submission_result_group in enumerate(SubmissionResultGroup.objects.filter(competition=competition)):
+            logger.info("Adding a submission result group.")
+            result_group_key = submission_result_group.key
+            leaderboards_dict[result_group_key] = dict()
+            leaderboards_dict[result_group_key]['label'] = submission_result_group.label
+            leaderboards_dict[result_group_key]['rank'] = submission_result_group.ordering
+            # Columns dictionary
+            columns_dictionary = dict()
+            # for index_score_def, submission_score_def in enumerate(SubmissionScoreDef.objects.filter(competition=competition)):
+            for index_score_def, submission_score_def_group in enumerate(SubmissionScoreDefGroup.objects.filter(group=submission_result_group)):
+                logger.info("Adding a submission score def group.")
+                score_def_key = submission_score_def_group.scoredef.key
+                columns_dictionary[score_def_key] = dict()
```

Comment:
this pattern is replicated pretty often, instead of this:

```python
new_dict = dict()
new_dict["thing"] = "stuff"
new_dict["other_thing"] = "other stuff"
```

Just do

```python
new_dict = {
    "thing": "stuff",
    "other_thing": "other stuff",
}
```